### PR TITLE
Move dependencies with common version numbers to Gradle Version Catalog

### PR DESCRIPTION
**Pull request by nprzy**
Gradle Version Catalog is a relatively new Gradle feature that makes it easier to manage Gradle versions across projects. That functionality honestly isn't that interesting for a project like this, but what is interesting is that Dependabot supports Gradle Version Catalogs. This moves the common version configurations out of `gradle.properties` into the Gradle Version Catalog.

For the other dependencies that don't share a common version number with each other, I opted to leave them as-is in `build.gradle.kts` for simplicity.

This PR also upgrades a few of the dependencies that had been getting neglected by Dependabot due to its lack of support for upgrading gradle.properties. And it also removes a few dependencies that I noticed weren't being used.

https://github.blog/changelog/2023-03-13-dependabot-version-updates-keeps-gradle-version-catalogs-up-to-date/

https://medium.com/@vh.dev/dependabot-in-action-d9b56b2be86c

### CODE REVIEW

**Comment by dvankley** [comment]
Whoops, looks like I created a conflict here by merging dependabot updates first.

**Comment by nprzy** [comment]
Rebased

**Comment by nprzy** [comment]
tl;dr: I think something may be wrong with your cache.

I wasn't able to reproduce. Digging into the error message, it's telling us that [this method](https://github.com/ktorio/ktor/blob/2.3.12/ktor-shared/ktor-serialization/ktor-serialization-jackson/jvm/src/JacksonConverter.kt#L187) from the `ktor-serialization-jackson` package does not exist. This PR updates this package from v2.1.1 to v2.3.12.

We can see from [this link to the v2.3.12 of this file](https://github.com/ktorio/ktor/blob/2.3.12/ktor-shared/ktor-serialization/ktor-serialization-jackson/jvm/src/JacksonConverter.kt#L187) that it indeed does exist. However, we can also see in your error message that it's not getting that class from the v2.3.12 jar, it's coming from the v2.1.1 jar. That method does [not exist in 2.1.1](https://github.com/ktorio/ktor/blob/2.1.1/ktor-shared/ktor-serialization/ktor-serialization-jackson/jvm/src/JacksonConverter.kt). Something similar is there, but with a slightly different signature, which we see exists but has been deprecated by v2.3.12.

The Kotlin compiler seems to be seeing the v2.3.12 jar and compiling the Firefly API client to use the new v2.3.12 method signature, but for some reason your classpath still appears to have the v2.1.1 jar on it.  I suspect if you kill your Gradle/Kotlin daemon processes and/or clear your build cache it should probably fix things.

**Comment by dvankley** [comment]
Thanks for looking into that; didn't have time to dig in much this morning. I wiped my gradle cache as you suggested and that seems to have fixed it.

**Comment by dvankley** [comment]
Other than that, looks good to go.
